airflow/providers/microsoft/azure/secrets/azure_key_vault.py

```diff
@@ -0,0 +1,131 @@
+    if you provide ``{"variables_prefix": "airflow-variables"}`` and request variable key ``hello``.
+
+    :param vault_url: The URL of an Azure Key Vault to use
+    :type vault_url: str
+    :param connections_prefix: Specifies the prefix of the secret to read to get Connections
+    :type connections_prefix: str
+    :param variables_prefix: Specifies the prefix of the secret to read to get Variables
+    :type variables_prefix: str
+    :param sep: separator used to concatenate secret_prefix and secret_id. Default: "-"
+    :type sep: str
+    """
+
+    def __init__(
+            self,
+            connections_prefix: str = 'airflow-connections',
+            variables_prefix: str = 'airflow-variables',
+            vault_url: str = '',
+            sep: str = '-',
+            **kwargs
+    ):
+        super().__init__()
+        self.vault_url = vault_url
+        self.connections_prefix = connections_prefix.rstrip(sep)
+        self.variables_prefix = variables_prefix.rstrip(sep)
+        self.sep = sep
+        self.kwargs = kwargs
+
+    @cached_property
+    def client(self):
+        """
+        Create a Azure Key Vault client.
+        """
+        credential = DefaultAzureCredential()
+        client = SecretClient(
+            vault_url=self.vault_url,
+            credential=credential,
+            **self.kwargs
+        )
+        return client
+
+    def get_conn_uri(self, conn_id: str) -> Optional[str]:
+        """
+        Get an Airflow Connection URI from an Azure Key Vault secret
+
+        :param conn_id: The Airflow connection id to retrieve
+        :type conn_id: str
+        """
+        name = self.build_path(self.connections_prefix, conn_id, self.sep)
+        return self.get_secret_value(name)
+
+    def get_variable(self, key: str) -> Optional[str]:
+        """
+        Get an Airflow Variable from an Azure Key Vault secret.
+
+        :param key: Variable Key
+        :type key: str
+        :return: Variable Value
+        """
+        name = self.build_path(self.variables_prefix, key, self.sep)
+        return self.get_secret_value(name)
+
+    @staticmethod
+    def build_path(path_prefix: str, secret_id: str, sep: str = '-') -> str:
+        """
+        Given a path_prefix and secret_id, build a valid secret name for the Azure Key Vault Backend.
+        Also replaces underscore in the path with dashes to support easy switching between
+        environmentvariables, so ``connection_default`` becomes ``connection-default``.
+
+        :param path_prefix: The path prefix of the secret to retrieve
+        :type path_prefix: str
+        :param secret_id: Name of the secret
+        :type secret_id: str
+        :param sep: Separator used to concatenate path_prefix and secret_id
+        :type sep: str
+        """
+        path = f'{path_prefix}{sep}{secret_id}'
+        return path.replace('_', sep)
+
+    def get_secret_value(self, name: str) -> Optional[str]:
```

Comment:
I've refactored _get_secret to get_secret_value to allow use of this backend for secrets that do not conform to the prefixes provided for connections and variables. We have a use case for this as secret names are sometimes internally created by another department that conform to different naming standards.

Reply:
We will have hooks for that, for example: https://github.com/apache/airflow/blob/master/airflow/providers/hashicorp/hooks/vault.py

WDYT ?

Reply:
I will make it a private method again and will create a hook later one.